cmd/utils/flaggroup.go

```diff
@@ -0,0 +1,319 @@
+		Flags: []cli.Flag{
+			DstDbTypeFlag,
+			DstDataDirFlag,
+			DstSingleDBFlag,
+			DstLevelDBCompressionTypeFlag,
+			DstNumStateTrieShardsFlag,
+			DstDynamoDBTableNameFlag,
+			DstDynamoDBRegionFlag,
+			DstDynamoDBIsProvisionedFlag,
+			DstDynamoDBReadCapacityFlag,
+			DstDynamoDBWriteCapacityFlag,
+		},
+	},
+	{
+		Name: "STATE",
+		Flags: []cli.Flag{
+			StateDBCachingFlag,
+			TrieMemoryCacheSizeFlag,
+			TrieBlockIntervalFlag,
+			TriesInMemoryFlag,
+		},
+	},
+	{
+		Name: "CACHE",
+		Flags: []cli.Flag{
+			CacheTypeFlag,
+			CacheScaleFlag,
+			CacheUsageLevelFlag,
+			MemorySizeFlag,
+			CacheWriteThroughFlag,
+			TxPoolStateCacheFlag,
+			TrieCacheLimitFlag,
+		},
+	},
+	{
+		Name: "CONSENSUS",
+		Flags: []cli.Flag{
+			ServiceChainSignerFlag,
+			RewardbaseFlag,
+		},
+	},
+	{
+		Name: "NETWORKING",
+		Flags: []cli.Flag{
+			BootnodesFlag,
+			ListenPortFlag,
+			SubListenPortFlag,
+			MultiChannelUseFlag,
+			MaxConnectionsFlag,
+			MaxPendingPeersFlag,
+			TargetGasLimitFlag,
+			NATFlag,
+			NoDiscoverFlag,
+			RWTimerWaitTimeFlag,
+			RWTimerIntervalFlag,
+			NetrestrictFlag,
+			NodeKeyFileFlag,
+			NodeKeyHexFlag,
+			NetworkIdFlag,
+			BaobabFlag,
+			CypressFlag,
+		},
+	},
+	{
+		Name: "METRICS",
+		Flags: []cli.Flag{
+			MetricsEnabledFlag,
+			PrometheusExporterFlag,
+			PrometheusExporterPortFlag,
+		},
+	},
+	{
+		Name: "VIRTUAL MACHINE",
+		Flags: []cli.Flag{
+			VMEnableDebugFlag,
+			VMLogTargetFlag,
+			VMTraceInternalTxFlag,
+		},
+	},
+	{
+		Name: "API AND CONSOLE",
+		Flags: []cli.Flag{
+			RPCEnabledFlag,
+			RPCListenAddrFlag,
+			RPCPortFlag,
+			RPCCORSDomainFlag,
+			RPCVirtualHostsFlag,
+			RPCApiFlag,
+			IPCDisabledFlag,
+			IPCPathFlag,
+			WSEnabledFlag,
+			WSListenAddrFlag,
+			WSPortFlag,
+			WSApiFlag,
+			WSAllowedOriginsFlag,
+			GRPCEnabledFlag,
+			GRPCListenAddrFlag,
+			GRPCPortFlag,
+			JSpathFlag,
+			ExecFlag,
+			PreloadJSFlag,
+			MaxRequestContentLengthFlag,
+		},
+	},
+	{
+		Name:  "LOGGING AND DEBUGGING",
+		Flags: debug.Flags,
+	},
+	{
+		Name: "SERVICECHAIN",
+		Flags: []cli.Flag{
+			ChildChainIndexingFlag,
+			MainBridgeFlag,
+			MainBridgeListenPortFlag,
+			SubBridgeFlag,
+			SubBridgeListenPortFlag,
+			AnchoringPeriodFlag,
+			SentChainTxsLimit,
+			ParentChainIDFlag,
+			VTRecoveryFlag,
+			VTRecoveryIntervalFlag,
+			ServiceChainAnchoringFlag,
+			ServiceChainNewAccountFlag,
+			KASServiceChainAnchorFlag,
+			KASServiceChainAnchorPeriodFlag,
+			KASServiceChainAnchorUrlFlag,
+			KASServiceChainAnchorOperatorFlag,
+			KASServiceChainAccessKeyFlag,
+			KASServiceChainSecretKeyFlag,
+			KASServiceChainXChainIdFlag,
+		},
+	},
+	{
+		Name: "MISC",
+		Flags: []cli.Flag{
+			GenKeyFlag,
+			WriteAddressFlag,
+			AutoRestartFlag,
+			RestartTimeOutFlag,
+			DaemonPathFlag,
+		},
+	},
+}
+
+// CategorizeFlags classified each flag into pre-defined flagGroups.
+func CategorizeFlags(flags []cli.Flag) []FlagGroup {
+	flagGroupsMap := make(map[string][]cli.Flag)
```

Comment:
I could do that, but I afraid repeating too many `for` loops. If `flagGroups` is directly used, a flag should iterate `flagGroups` slice to find its group the flag is classified as. Since `CategorizeFlags` already has a loop repeating `for` 3 times, I introduced `flagGroupsMap` converting it to `flagGroups` outside of the repeated `for` loop.